cmd/antrea-agent/agent.go

```diff
@@ -298,9 +299,20 @@ func run(o *Options) error {
 
 	// Initialize flow exporter to start go routines to poll conntrack flows and export IPFIX flow records
 	if features.DefaultFeatureGate.Enabled(features.FlowExporter) {
+		match, err := regexp.MatchString("\\[.*\\]:.*", o.config.FlowCollectorAddr)
+		if err != nil {
+			return fmt.Errorf("Failed to parse FlowCollectorAddr: %s", o.config.FlowCollectorAddr)
+		}
+		svcCIDR := serviceCIDRNet
+		addrFamily := "ipv4"
+		if match {
+			svcCIDR = serviceCIDRNetv6
+			addrFamily = "ipv6"
+		}
```

Comment:
is the following scenario not valid?
* the cluster is dual-stack, and Nodes have both an IPv4 address and an IPv6 address
* the primary address family for the cluster (serviceCIDR) is IPv6
* the collector runs outside the cluster (public IP) and only has an IPv4 address

Basically I'm asking whether it makes sense to use `o.config.FlowCollectorAddr` to determine the service address family? Why can't we just check whether `serviceCIDRNetv6 != ""`?

Reply:
Thank you for the question. I think if user want to use flow exporter, `FlowCollectorAddr` should be correctly configured. Also, with AntreaProxy enabled, `serviceCIDRNetv6` is not needed.

Reply:
`serviceCIDRNetv6` is not needed with AntreaProxy enabled while `FlowCollectorAddr` should be correctly configured by users.

Reply:
That only answers part of my question though.

What if my flow collector is running out-of-cluster and only has a public IPv4 address, but my cluster is dual-stack?

Reply:
Sorry, I only paid attention to the sentences after "Basically"...
You are right. flowCollectorAddr is unreliable. Instead, I use nodeAddr now for judgement. Then in that situation, address family is IPv6 and FlowCollectorAddr is IPv4.

Reply:
Does this mean we support pod-to-externalTraffic where pod is of IPv6 and external IP is IPv4? Will the IPv6 address be SNATed to IPv4 address?

Reply:
Maybe not now. I created an ipfix-collector in an IPv6-only cluster. And I hard-coded its address into e2e test code in a dual-stack cluster, with IPv4 service CIDR. I ran the flow exporter test and checked logs of ipfix-collector. There's no IPv4 record.

Reply:
In the above test, could antrea-agent pod (flow exporter) with IPv6 make a connection to the flow collector with IPv4 address? Wondering if that communication is possible at all?

Reply:
I guess you mean antrea-agent pod with IPv4 as PodIP to connect flow collector pod with IPv6? The traffic is ok.

Reply:
Please ignore my comment. Understood the scenario in Antonin's comment after offline discussion.

Reply:
@srikartati Do you think this PR can be merged? The latest code uses `NodeAddr` to decide if in IPv4 or IPv6.